src/libYARP_OS/CMakeLists.txt

```diff
@@ -404,9 +404,18 @@ target_include_directories(YARP_OS PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOUR
 target_link_libraries(YARP_OS PRIVATE ${ACE_LIBRARIES}
                               PUBLIC YARP::YARP_conf)
 
-target_compile_features(YARP_OS PUBLIC cxx_nullptr
-                                       cxx_override
-                                       cxx_rvalue_references)
+if(NOT CMAKE_MINIMUM_REQUIRED_VERSION VERSION_LESS 3.1)
+  message(AUTHOR_WARNING "CMAKE_MINIMUM_REQUIRED_VERSION is now ${CMAKE_MINIMUM_REQUIRED_VERSION}. This check can be removed.")
+endif()
+if(CMAKE_VERSION VERSION_LESS 3.1)
+  if(DEFINED CXX11_FLAGS)
+    target_compile_options(YARP_OS PUBLIC ${CXX11_FLAGS})
+  endif()
+else()
+  target_compile_features(YARP_OS PUBLIC cxx_nullptr
+                                         cxx_override
+                                         cxx_rvalue_references)
+endif()
```

Comment:
btw, not sure if in cmake 3.8 or 3.9 there is the `cxx_std_11` feature that simplifies this... I don't think it's worth to have an extra `if` here though...